airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -498,6 +581,17 @@ def read_pod_logs(
             post_termination_timeout=post_termination_timeout,
         )
 
+    @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(), reraise=True)
+    def get_container_names(self, pod: V1Pod) -> list[str]:
+        """Return container names from the POD except for the airflow-xcom-sidecar container."""
+        containers = []
+        pod_info = self.read_pod(pod)
+        for container_spec in pod_info.spec.containers:
+            if container_spec.name != ContainerNames.XCOM_CONTAINER:
+                containers.append(container_spec.name)
+
+        return containers
```

Comment:
This seems to be a simpler equivalent

```suggestion
        pod_info = self.read_pod(pod)
        return [
            container_spec.name
            for container_spec in pod_info.spec.containers
            if container_spec.name != ContainerNames.XCOM_CONTAINER
        ]
```

Reply:
Sure. Made the changes here